File: 10_minute_challenges/Space_invaders_p5.js/ship.js
function Ship() {
  this.x = width/2;
  this.xdir = 0;

  this.show = function() {
    fill(255);   
    rectMode(CENTER); 
    rect(this.x, height-20, 20, 60);
  }

  this.setDir = function(dir) {
    this.xdir = dir;
  }

  this.move = function(dir) {
    this.x += this.xdir*5;
  }

}

File: CodingChallenges/CC_05_Space_invaders_py/Ship.py
# Daniel Shiffman - Translated  to Python by Ali Al Saqban
# http://codingtra.in
# http://patreon.com/codingtrain

# Space Invaders
# https://youtu.be/biN3v3ef-Y0

class Ship(object):
    def __init__(self):
        self.x = width/2
        self.xdir = 0
    
    def show(self):
        fill(255)
        rectMode(CENTER)
        rect(self.x, height-20, 20, 60)
    
    def setDir(self,dir):
        self.xdir = dir
    
    def move(self):
        self.x += self.xdir*5
    